src/iarc7_motion/iarc_tasks/height_holder.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+# A helper class for a task that will spit out a velocity that's appropriate
+# to reach and hold a desired position in the z dimension
+# Uses a simple p-controller
+
+import math
+import rospy
+
+from geometry_msgs.msg import TwistStamped
+from nav_msgs.msg import Odometry
+
+class HeightHolder():
+    def __init__(self):
+        self._k_z = .1 #p-term for z-height
+        self._odometry = None
+        self._current_velocity_sub = rospy.Subscriber('/odometry/filtered',
+                                              Odometry,
+                                              self._current_velocity_callback)
+        try:
+            self._TRACK_HEIGHT = rospy.get_param('~track_roomba_height')
+        except KeyError as e:
+            rospy.logerr('Could not lookup a parameter for track roomba task')
+            raise
+
+        while self._odometry is None:
+            pass
+
+    #uses a p-controller to return a velocity to maintain a height
+    #that is set as the param _track_roomba_height
+    def get_height_hold_response(self):
+        delta_z = self._TRACK_HEIGHT - self._odometry.pose.pose.position.z 
```

Comment:
Because there are two threads modifying the state of variables in this object we should probably use locks at the start of this function and current velocity callback.